src/intelli/node/objectsink.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * GTlab IntelliGraph
+ *
+ *  SPDX-License-Identifier: BSD-3-Clause
+ *  SPDX-FileCopyrightText: 2024 German Aerospace Center
+ *
+ *  Author: Jens Schmeink <jens.schmeink@dlr.de>
+ */
+#include "objectsink.h"
+
+#include <intelli/data/object.h>
+
+#include <QPushButton>
+
+#include "gt_application.h"
+#include "gt_datamodel.h"
+#include "gt_project.h"
+
+using namespace intelli;
+
+ObjectSink::ObjectSink() :
+    Node("Object sink"),
+    m_target("target", tr("Target"), tr("Target"), QString(), this,
+               QStringList() << GT_CLASSNAME(GtObject), true)
+{
+    m_in  = addInPort({typeId<ObjectData>(), tr("Object")});
+
+    registerProperty(m_target);
+
+    // registering the widget factory
+    registerWidgetFactory([=](intelli::Node& /*this*/){
+        auto w = std::make_unique<QPushButton>("Export");
+
```

Comment:
Button can be triggered if its not connected yet resulting in a null-pointer dereference in `doExport`
```suggestion
w->setEnabled(false);

```